homeassistant/components/websocket_api/http.py

```diff
@@ -71,17 +73,35 @@ def __init__(self, hass: HomeAssistant, request: web.Request) -> None:
     async def _writer(self) -> None:
         """Write outgoing messages."""
         # Exceptions if Socket disconnected or cancelled by connection handler
+        to_write = self._to_write
+        logger = self._logger
+        wsock = self.wsock
         with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
             while not self.wsock.closed:
-                if (process := await self._to_write.get()) is None:
+                if (process := await to_write.get()) is None:
+                    break
+                message = process if isinstance(process, str) else process()
+
+                if not self.command_phase or to_write.empty():
+                    logger.debug("Sending %s", message)
+                    await wsock.send_str(message)
+                    continue
+
+                messages: list[str] = [message]
+                while not to_write.empty():
+                    if (process := to_write.get_nowait()) is None:
```

Comment:
When it's None we're shutting down. Should we still send all other messages?

Reply:
Seems to work fine. Assuming non of the tests break, thats a much better design

e7510cb21d78838dea136db0423d218d758398d9